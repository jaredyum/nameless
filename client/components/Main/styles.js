export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%',
    [theme.breakpoints.up('md')]: {
      maxWidth: 'calc(100% - 240px)'
    }
  },
  toolbar: theme.mixins.toolbar
});
