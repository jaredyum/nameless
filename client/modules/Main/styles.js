export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column'
  },
  toolbar: theme.mixins.toolbar
});
