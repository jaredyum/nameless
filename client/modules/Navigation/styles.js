/**
 * The default width of the drawer (sidebar).
 * @const {number}
 */
const DRAWER_WIDTH = 240;

/**
 * The parent styles for this modules.
 * @param {!Object} theme The MUI theme object.
 * @return {!Object.<string, !Object>} The generated styles.
 */
export default theme => ({
  active: {
    fontWeight: 'bold'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  drawerPaper: {
    width: DRAWER_WIDTH,
    [theme.breakpoints.up('md')]: {
      position: 'relative'
    }
  },
  flex: {
    flex: 1
  },
  link: {
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none',
    padding: '5px 0'
  },
  menuItem: {
    height: 'auto',
    padding: 0,
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none'
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  },
  toolbar: {
    backgroundColor: theme.palette.primary.main,
    color: '#fff',
    ...theme.mixins.toolbar
  }
});
