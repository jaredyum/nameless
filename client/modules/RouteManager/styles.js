export default {
  routeManager: {
    flex: 1,
    display: 'flex'
  }
};
