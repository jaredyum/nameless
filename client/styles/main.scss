// Vendors
@import './vendors/normalize';
@import './vendors/reset';

// Modules
@import './modules/defaults';
