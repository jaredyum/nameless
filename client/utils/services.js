/* eslint-disable no-console */
import axios from 'axios';

// Firebase config.
import firebase from 'services/firebase';

// Errors
import { API_ERRORS } from 'copy/Global/errors';

/**
 * Decorates the headers for a request.
 * @param {string} token The firebase token.
 * @return {!Object} The decorated headers object.
 */
const decorateHeaders = token => ({
  Authorization: `Bearer ${token}`,
  'Content-Type': 'application/json'
});

export const defaultCatch = (endPoint, source = 'services') => (e) => {
  console.error(`${source}/${endPoint}`, e);
  throw e;
};

/**
 * Extracts and parses the Firebase token to use for requests.
 * @return {!Promise} A promise resolving with the firebase token.
 */
export const getUserToken = () => firebase
  .then(auth => auth.currentUser)
  .then(user => user.getIdToken())
  .catch(defaultCatch('getUserToken'));

/**
 * Request wrapper for all request types (GET, POST, PUT, DELETE, etc).
 * @param {!Object} config The request configuration.
 * @param {string} config.url The URL for the request.
 * @param {string} config.method The method (GET, POST, etc) for the request.
 * @param {!Object} config.payload Optional request payload.
 * @return {!Promise} Promise containing the axios response.
 */
export const makeRequest = ({
  url,
  method = 'get',
  payload = null
}) => getUserToken()
  .then(token => ({
    headers: decorateHeaders(token),
    method,
    url,
    ...(
      payload ? { data: payload } : {}
    )
  }))
  .then(axios)
  .then((res) => {
    if (res && (res.status >= 200 || res.status < 400)) {
      return res;
    }
    throw new Error(API_ERRORS.BAD_REQUEST);
  })
  .then((res) => {
    if ('location' in res.headers) {
      return makeRequest(res.headers.location, 'GET', null, true);
    }

    return res;
  });
