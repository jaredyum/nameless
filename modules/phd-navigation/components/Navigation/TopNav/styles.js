export default theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  flex: {
    flex: 1
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  }
});
