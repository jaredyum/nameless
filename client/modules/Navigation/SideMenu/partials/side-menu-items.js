import React from 'react';

// React Router
import { NavLink } from 'react-router-dom';

// MUI Components
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';

// Local styles
import styles from '../../styles';

const SideMenuItems = ({ toggleHandler, classes, collection }) =>
  collection.map(({
    label,
    path,
    clickHandler
  }) => {
    // If this is an action item (NOT a link), then clicking should BOTH execute
    // the action AND close the menu.
    if (clickHandler) {
      const closeAndExec = () => {
        toggleHandler();
        clickHandler();
      };

      return (
        <ListItem onClick={closeAndExec} key={label}>
          <span className={classes.link}>{ label }</span>
        </ListItem>
      );
    }

    return (
      <ListItem key={label}>
        <NavLink
          to={path}
          className={classes.link}
          onClick={toggleHandler}
          replace
        >{ label }</NavLink>
      </ListItem>
    );
  });

export const SideMenuItemsJest = SideMenuItems;

export default withStyles(styles)(SideMenuItems);
