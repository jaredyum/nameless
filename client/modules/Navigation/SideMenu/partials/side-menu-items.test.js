import { testCommonComponentAttrs } from 'utils/tests';

// Local styles
import styles from '../../styles';

// Components
import { SideMenuItemsJest } from './side-menu-items';

const toggleHandler = jest.fn();

const defaultProps = {
  classes: styles,
  collection: [{
    label: 'foo',
    path: '/foo'
  }, {
    label: 'bar',
    clickHandler: jest.fn()
  }],
  toggleHandler
};

describe('<MyComponent /> Component', () => {
  testCommonComponentAttrs(SideMenuItemsJest, defaultProps);
});
