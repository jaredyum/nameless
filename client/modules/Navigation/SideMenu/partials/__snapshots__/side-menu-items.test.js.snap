// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> Component when rendering should render a component matching the snapshot 1`] = `
Array [
  <WithStyles(ListItem)
    key="foo"
  />,
  <WithStyles(ListItem)
    key="bar"
    onClick={[Function]}
  />,
]
`;
